MAINTAINERS.md

```diff
@@ -1,18 +1,122 @@
+2. The RM then proposes a “feature freeze” date (and intent to cut the branch).  This should be a date  *one week* in the future (and therefore one week before release).  During this week, all maintainers will: 
+	1. Mark “blocker” issues preventing a successful release
+	2. Merge all PRs that are ready into the main, or clearly label them as being for a future release.  
+	3. Review merged PRs and determine if they need a backport to the current release. 
+	4. If a maintainer is not sure if a merged PR needs a backport, they should post in the PR asking the submitter their intention. 
+3. 24 hours prior to feature freeze (cutting the branch):
+    1. RM update the forum post with a “final notice“ message to ensure all blockers are identified (some may be fixed by then).
+4. On day of feature freeze:
+    1. RM posts a “feature freeze” notice that the branching is underway and no more PRs should be merged.
+    2. RM branches from the .x branch into a new .m+1 branch. (.x is now the next minor)
+    3. RM sends a “branching complete” email that the branch has been cut and a reminder of the release date requesting all blockers be cleared in time of the release
+5. During the week between "feature freeze" and release, all maintainers will:
+	1. Will monitor issues looking for fixes that need to go in to the release. 
+5. 24 Hours prior to release:
+    1. RM sends a “final notice” for blockers email; also requesting a “speak now or forever hold your peace” on critical blockers. If something is holding up the release a postponed release may occur.
+6. On Day of Release:
+    1. RM tags the branch with the release version
+    2. RM begins building and signing artifacts
+    3. RM stages artifacts to maven repo 
+    4. an email is sent once artifacts are published requesting a quorum vote which includes committers run smoke tests on the staged artifacts;
+    5. RM sums votes 
+    6. RM publishes tested artifacts
+
+
+## Pull Requests (PRs)
+
+### PR Lifecycle
+
+Here's the life cycle of a PR:
+
+1. The developer should create a Github issue explaining the problem, steps to reproduce, potential fix(es).
+2. The developer creates a PR.  They should tag the related issue in the PR description
+3. The maintainer labels the PR (//TODO with what?), reviews it and decides if there is need for another pair of eyes on the PR. e.g. Some PRs may warrant someone from an area of expertise to review it. It could also be based on complexity, security, or breaking changes.
+4.  If all looks good, maintainer merges the PR. If the PR changes some behavior that should be added/updated in end user documentation, the maintainer should add documentation pending label to the issue and keep it open. Remove the label once documentation PR is added and merged.
```

Comment:
We do squash merge, should call that out (I don't like it).

Reply:
I have recently noticed the quality of commit messages is not great, opened a [forum issue](https://discuss.opendistrocommunity.dev/t/missing-commit-messages-and-changelogs/) about this, but perhaps we could add in this section that there should always be a descriptive commit message? 
There is a stark difference comparing commit messages prior to the fork to the ones in `main` since the fork.

Reply:
I will open an issue for this to be added in `CONTRIBUTING.md` for the PR authors side, but maybe here we should have the reviewer also consider the commit message as part of the PR.

Reply:
opened https://github.com/opensearch-project/OpenSearch/issues/851 in reference to my comment